src/main/java/duke/TaskList.java

```diff
@@ -0,0 +1,203 @@
+        }
+        String subject = deadlinetokens[0];
+        String deadline = deadlinetokens[1];
+        list.add(new Deadline(subject, deadline));
+
+    }
+    /**
+     * Creates an todo task in the tasklist.
+     * @param str The string representation of the task.
+     */
+    public static void todoCase(String str) throws DukeException {
+        str = str.replace("todo", "");
+        int strcount = str.split("\\s").length;
+
+        if(strcount == 1) {
+            throw new DukeException("OOPS!!! The description of a todo cannot be empty. " +
+                    " Please give this instruction in the following format: todo [description]");
+        }
+        list.add(new Task(str));
+
+    }
+    /**
+     * marks a task as done.
+     * @param  tokens The command to mark task as done.
+     */
+    public static void markCase(String[] tokens) throws DukeException{
+        if(tokens.length != 2) {
+            throw new DukeException("please give this instruction in the following format: mark [task number]");
+        }
+        int no = Integer.parseInt(tokens[1]) - 1;
+        list.get(no).markAsDone();
+        System.out.println("Nice! I've marked this task as done:\n");
+        System.out.println(list.get(no).ToString());
+        store.save();
+    }
+
+    /**
+     * marks a task as undone.
+     * @param  tokens The command to mark task as undone.
+     */
+    public static void unmarkCase(String[] tokens) throws DukeException{
+        if(tokens.length != 2) {
+            throw new DukeException("please give this instruction in the following format: unmark [task number]");
+        }
+        int no = Integer.parseInt(tokens[1]) - 1;
+        list.get(no).unmarkAsDone();
+        System.out.println("OK, I've marked this task as not done yet:");
+        System.out.println(list.get(no).ToString());
+        store.save();
+    }
+
+    /**
+     * removes a task.
+     * @param  tokens The command to remove a task.
+     */
+    public static ArrayList<Task> removeCase(String[] tokens) throws DukeException{
+        if(tokens.length != 2) {
+            throw new DukeException("please give this instruction in the following format: delete [task number]");
+        }
+        System.out.println("Noted. I've removed this task:\n");
+        int no = Integer.parseInt(tokens[1])-1;
+        System.out.println(list.get(no).ToString());
+        list.remove(no);
+
+
+
+        System.out.println("Now you have " + list.size() + " tasks in the list.");
+        store.save();
+
+        return list;
+
+
+
+    }
+
+    public static void findCase(String str) throws DukeException{
+        if(str.equals("find")) {
+            throw new DukeException("please give this instruction in the following format: find [keyword]");
+        }
+        String keyword = str.replace("find", "");
+        System.out.println("Here are the matching tasks in your list:");
+        for (int i = 0; i < list.size(); i++) {
+            int n = 0;
```

Comment:
Your variable name should be more detailed. I assume this counts the number of tasks containing the keywords? 

```suggestion
            int caseCount = 0;
```